### ISSUE TITLE
Include asset manifest in created zip package?

### ISSUE THREAD

**Issue by Tongboy** [alienfast/elastic-beanstalk#25]
Hi,

I've been loving the gem!

Like most I've done some experimenting with reducing deployment time and overhead on the EB servers. The simplest & most impactful change i've made is performing the asset compilation on my dev or CI server and including the compiled assets in the zip package to get deployed to EB. the unintended benefit beyond reduced deployment time is a large reduction in memory consumption during deployment - by removing the need to compile the assets on the server I can deploy to a t2.micro instance without a problem where as with asset compilation on EB at minimum my app required a t2.medium instance.

I ran in to an issue with this gem when performing that operation - all of the assets pack in nicely but the asset manifest file (/public/assets/.sprockets-manifest-[hash].json is excluded. my current deploy script includes a line to add it to the created zip file between eb package & eb deploy.

A valid workaround is to specify a different filename for the manifest file (which sprockets config supports.) What are your thoughts as to the best method - include the default sprockets manifest file in the package if it exists or should I rename the manifest file?

Thank you,

**Comment by rosskevin** [comment]
See [config.rb](https://github.com/alienfast/elastic-beanstalk/blob/master/lib/elastic/beanstalk/config.rb#L34).  There are default package includes, but I'm guessing that [FileList](https://github.com/alienfast/elastic-beanstalk/blob/master/lib/elastic/beanstalk/tasks/eb.rake#L232) is ignoring files starting with a leading `.`

```yaml
package:
    includes:  ['public/assets/.sprockets-manifest-*.json']
```

This should do it.  Strangely, I'm deploying precompiled assets without this manifest and all seems well.  What error are you seeing?  Can you post a stack trace?

**Comment by Tongboy** [comment]
ah, thank you, I missed the includes option for the package! That works perfectly!

I deployed without the manifest file to an instance to try and reproduce my issue... I swear I used to get a stack trace but that could have been on an older EB platform than we're using now. Now I just get 404's returned for assets that don't have hashes on them. When I include the manifest file everything works the way it should.

This happens whether I configure rails to serve static assets or not. I'm running Puma rather than passenger. Maybe the difference is somewhere in there?

Thank you!
